instrumentation/vertx-reactive-3.5/javaagent/src/test/groovy/VertxReactivePropagationTest.groovy

```diff
@@ -85,5 +95,76 @@ class VertxReactivePropagationTest extends AgentInstrumentationSpecification {
     }
   }
 
+  def "should propagate context correctly over vert.x rx-java framework with high concurrency"() {
+    setup:
+    int count = 100
+    def baseUrl = "http://localhost:$port/listProducts"
+    def latch = new CountDownLatch(1)
+
+    def pool = Executors.newFixedThreadPool(8)
+    def propagator = GlobalOpenTelemetry.getPropagators().getTextMapPropagator()
+    def setter = { Request.Builder carrier, String name, String value ->
+      carrier.header(name, value)
+    }
+
+    when:
+    count.times { index ->
+      def job = {
+        latch.await()
+        Request.Builder builder = new Request.Builder().url("$baseUrl?$TEST_REQUEST_ID_PARAMETER=$index").get()
+
+        runUnderTrace("client " + index) {
+          Span.current().setAttribute(TEST_REQUEST_ID_ATTRIBUTE, index)
+          propagator.inject(Context.current(), builder, setter)
```

Comment:
Why do we need this manual propagation? Looks very suspicious to me...

Reply:
The request is performed with OkHttp, which is not the library being tested here (as this is a server test), thus there is no `testInstrumentation` for it. This is the same logic used in `HttpServerTest`. This approach still makes sure the server instrumentation respects propagated headers without requiring instrumentation of the client.